src/hipscat/inspection/almanac.py

```diff
@@ -0,0 +1,175 @@
+
+    def __init__(self, include_default_dir=True, dirs=None):
+        """Create new almanac"""
+        self.files = {}
+        self.entries = {}
+        self.dir_to_catalog_name = {}
+        self._init_files(include_default_dir=include_default_dir, dirs=dirs)
+        self._init_catalog_objects()
+        self._init_catalog_links()
+
+    def _init_files(self, include_default_dir=True, dirs=None):
+        if include_default_dir:
+            default_dir = AlmanacCatalogInfo.get_default_dir()
+            if default_dir:
+                self._add_files_to_namespace(default_dir)
+        if pd.api.types.is_dict_like(dirs):
+            for key, value in dirs.items():
+                self._add_files_to_namespace(value, key)
+        elif pd.api.types.is_list_like(dirs):
+            self._add_files_to_namespace(dirs)
+        elif dirs is not None:
+            self._add_files_to_namespace(dirs)
+
+    def _add_files_to_namespace(self, directory, namespace=""):
+        """Get yaml files within a directory or list of directories."""
+        if not pd.api.types.is_list_like(directory):
+            directory = [directory]
+
+        files = []
+        for input_path in directory:
+            if os.path.isfile(input_path):
+                files.append(input_path)
+                continue
+
+            input_paths = glob.glob(f"{input_path}/**.yml")
+            input_paths.sort()
+            files.extend(input_paths)
+
+        if namespace in self.files:
+            self.files[namespace].extend(files)
+        else:
+            self.files[namespace] = files
+
+    def _init_catalog_objects(self):
+        for namespace, files in self.files.items():
+            for file in files:
+                catalog_info = AlmanacCatalogInfo.from_file(file)
+                catalog_info.namespace = namespace
+                if namespace:
+                    full_name = f"{namespace}:{catalog_info.catalog_name}"
+                else:
+                    full_name = catalog_info.catalog_name
+                if full_name in self.entries:
+                    raise ValueError(
+                        f"Duplicate catalog name ({full_name}). Try using namespaces."
+                    )
+                self.entries[full_name] = catalog_info
+                self.dir_to_catalog_name[catalog_info.catalog_path] = full_name
+
+    def _init_catalog_links(self):
+        for catalog_entry in self.entries.values():
+            if catalog_entry.catalog_type == CatalogType.OBJECT:
+                pass
+            elif catalog_entry.catalog_type == CatalogType.SOURCE:
+                catalog_entry.primary_link = self._get_linked_catalog(
+                    catalog_entry.get_primary_text(),
+                    "primary",
+                    "source",
+                    catalog_entry.catalog_name,
+                    catalog_entry.namespace,
+                )
+                if catalog_entry.primary_link:
```

Comment:
Should this if statement not also appear in the other cases when appending to the primary link?

Reply:
I think it's just the association that's missing it, is that on purpose?